### PR TITLE
Porting System.IO.Packaging source comments to Docs

### PR DESCRIPTION
**Pull request by carlossanlop**
## Summary

Ported source code comments from System.IO.Packaging.
Note: Just like in my previous pull request for System.IO.Compression, there were a couple of parameters that did not have the same name in the ref and in the src. I won't modify the names, will respect what reflection and ref had. We can decide what to change on the source code comments later.

Fixes #Issue_Number (if available)

### CODE REVIEW

**Comment by carlossanlop** [comment]
@mairaw Please review this whenever you get a chance. Thank you!
@JeremyKuhne I am assuming you are the owner of this namespace since you show up [here](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/issue-guide.md) as the owner of System.IO (Packaging in particular doesn't show up). If you are not the owner, let me know.

File: xml/System.IO.Packaging/PackUriHelper.xml [hunk 1]
```diff
@@ -475,6 +475,8 @@
         <altmember cref="M:System.IO.Packaging.PackUriHelper.ResolvePartUri(System.Uri,System.Uri)" />
         <related type="ExternalDocumentation" href="https://go.microsoft.com/fwlink/?LinkID=160055">Writing a Package Sample</related>
         <related type="ExternalDocumentation" href="https://go.microsoft.com/fwlink/?LinkID=71255">Open Packaging Conventions Specification</related>
+        <exception cref="T:System.ArgumentNullException">If the <paramref name="partUri" /> parameter is <see langword="null" />.</exception>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/PackUriHelper.xml]:
```suggestion
        <exception cref="T:System.ArgumentNullException"><paramref name="partUri" /> is <see langword="null" />.</exception>
```

File: xml/System.IO.Packaging/PackUriHelper.xml [hunk 2]
```diff
@@ -475,6 +475,8 @@
...
+        <exception cref="T:System.ArgumentException">If the <paramref name="partUri" /> parameter is not an absolute <see cref="T:System.Uri" />.</exception>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/PackUriHelper.xml]:
```suggestion
        <exception cref="T:System.ArgumentException"><paramref name="partUri" /> is not an absolute <see cref="T:System.Uri" />.</exception>
```

File: xml/System.IO.Packaging/PackUriHelper.xml [hunk 3]
```diff
@@ -523,6 +525,7 @@
         <altmember cref="M:System.IO.Packaging.PackUriHelper.GetPartUri(System.Uri)" />
         <altmember cref="M:System.IO.Packaging.PackUriHelper.GetPackageUri(System.Uri)" />
         <related type="ExternalDocumentation" href="https://go.microsoft.com/fwlink/?LinkID=71255">Open Packaging Conventions Specification</related>
+        <exception cref="T:System.ArgumentException">If the <paramref name="partUri" /> parameter does not have a valid <see cref="T:System.Uri" /> syntax.</exception>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/PackUriHelper.xml]:
```suggestion
        <exception cref="T:System.ArgumentException"><paramref name="partUri" /> does not have a valid <see cref="T:System.Uri" /> syntax.</exception>
```

File: xml/System.IO.Packaging/PackUriHelper.xml [hunk 4]
```diff
@@ -570,6 +573,8 @@
         <altmember cref="Overload:System.IO.Packaging.PackUriHelper.Create" />
         <altmember cref="M:System.IO.Packaging.PackUriHelper.GetPartUri(System.Uri)" />
         <related type="ExternalDocumentation" href="https://go.microsoft.com/fwlink/?LinkID=71255">Open Packaging Conventions Specification</related>
+        <exception cref="T:System.ArgumentNullException">If the <paramref name="packUri" /> parameter is <see langword="null" />.</exception>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/PackUriHelper.xml]:
```suggestion
        <exception cref="T:System.ArgumentNullException"><paramref name="packUri" /> is <see langword="null" />.</exception>
```

File: xml/System.IO.Packaging/PackUriHelper.xml [hunk 5]
```diff
@@ -570,6 +573,8 @@
...
+        <exception cref="T:System.ArgumentException">If the <paramref name="packUri" /> parameter is not an absolute <see cref="T:System.Uri" />.</exception>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/PackUriHelper.xml]:
```suggestion
        <exception cref="T:System.ArgumentException"><paramref name="packUri" /> is not an absolute <see cref="T:System.Uri" />.</exception>
```

File: xml/System.IO.Packaging/PackUriHelper.xml [hunk 6]
```diff
@@ -733,6 +738,7 @@
           <paramref name="sourcePartUri" /> or <paramref name="targetPartUri" /> is <see langword="null" />.</exception>
         <altmember cref="M:System.IO.Packaging.PackUriHelper.ResolvePartUri(System.Uri,System.Uri)" />
         <related type="ExternalDocumentation" href="https://go.microsoft.com/fwlink/?LinkID=71255">Open Packaging Conventions Specification</related>
+        <exception cref="T:System.ArgumentException">If either the <paramref name="sourcePartUri" /> or the <paramref name="targetPartUri" /> parameters do not have a valid <see cref="T:System.Uri" /> syntax.</exception>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/PackUriHelper.xml]:
```suggestion
        <exception cref="T:System.ArgumentException">Either the <paramref name="sourcePartUri" /> or <paramref name="targetPartUri" /> does not have a valid <see cref="T:System.Uri" /> syntax.</exception>
```

File: xml/System.IO.Packaging/PackageRelationshipSelector.xml [hunk 7]
```diff
@@ -119,6 +119,7 @@ The <paramref name="sourceUri" /> is not the <see cref="T:System.IO.Packaging.Pa
         <returns>A list of relationships that match the selection parameters specified to the <see cref="M:System.IO.Packaging.PackageRelationshipSelector.#ctor(System.Uri,System.IO.Packaging.PackageRelationshipSelectorType,System.String)" /> constructor.</returns>
         <remarks>To be added.</remarks>
         <related type="ExternalDocumentation" href="https://go.microsoft.com/fwlink/?LinkId=74494">Digital Signing Framework of the Open Packaging Conventions</related>
+        <exception cref="T:System.ArgumentNullException">If the <paramref name="package" /> parameter is <see langword="null" />.</exception>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/PackageRelationshipSelector.xml]:
```suggestion
        <exception cref="T:System.ArgumentNullException"><paramref name="package" /> is <see langword="null" />.</exception>
```

File: xml/System.IO.Packaging/ZipPackage.xml [hunk 8]
```diff
@@ -78,12 +78,15 @@
         <Parameter Name="compressionOption" Type="System.IO.Packaging.CompressionOption" />
       </Parameters>
       <Docs>
-        <param name="partUri">To be added.</param>
-        <param name="contentType">To be added.</param>
-        <param name="compressionOption">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="partUri">The <see cref="T:System.Uri" /> that represents the <see cref="T:System.IO.Packaging.PackagePart" />.</param>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackage.xml]:
```suggestion
        <param name="partUri">The Uri that represents the <see cref="T:System.IO.Packaging.PackagePart" />.</param>
```

File: xml/System.IO.Packaging/ZipPackage.xml [hunk 9]
```diff
@@ -78,12 +78,15 @@
...
+        <param name="contentType">A string representing the content type of the <see cref="T:System.IO.Packaging.PackagePart" />.</param>
+        <param name="compressionOption">The compression option to be used for this <see cref="T:System.IO.Packaging.PackagePart" />.</param>
+        <summary>Custom implementation for the underlying file format. Adds a new item to the zip archive corresponding to the <see cref="T:System.IO.Packaging.PackagePart" /> in the <see cref="T:System.IO.Packaging.ZipPackage" />.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">If the <paramref name="partUri" /> parameter is <see langword="null" />.</exception>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackage.xml]:
```suggestion
        <exception cref="T:System.ArgumentNullException"><paramref name="partUri" /> is <see langword="null" />.</exception>
```

File: xml/System.IO.Packaging/ZipPackage.xml [hunk 10]
```diff
@@ -78,12 +78,15 @@
...
+        <exception cref="T:System.ArgumentException">If the <paramref name="partUri" /> parameter does not have a valid <see cref="T:System.Uri" /> syntax.</exception>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackage.xml]:
```suggestion
        <exception cref="T:System.ArgumentException"><paramref name="partUri" /> does not have a valid <see cref="T:System.Uri" /> syntax.</exception>
```

File: xml/System.IO.Packaging/ZipPackage.xml [hunk 11]
```diff
@@ -78,12 +78,15 @@
...
+        <exception cref="T:System.ArgumentOutOfRangeException">If the <paramref name="compressionOption" /> parameter does not have a valid <see cref="T:System.IO.Packaging.CompressionOption" /> value.</exception>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackage.xml]:
```suggestion
        <exception cref="T:System.ArgumentOutOfRangeException"><paramref name="compressionOption" /> does not have a valid <see cref="T:System.IO.Packaging.CompressionOption" /> value.</exception>
```

File: xml/System.IO.Packaging/ZipPackage.xml [hunk 12]
```diff
@@ -113,9 +116,11 @@
         <Parameter Name="partUri" Type="System.Uri" />
       </Parameters>
       <Docs>
-        <param name="partUri">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="partUri">The <see cref="T:System.Uri" /> that represents the <see cref="T:System.IO.Packaging.PackagePart" />.</param>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackage.xml]:
```suggestion
        <param name="partUri">The Uri that represents the <see cref="T:System.IO.Packaging.PackagePart" />.</param>
```

File: xml/System.IO.Packaging/ZipPackage.xml [hunk 13]
```diff
@@ -113,9 +116,11 @@
...
+        <summary>This method is for a custom implementation specific to the file format. Deletes the <see cref="T:System.IO.Packaging.PackagePart" /> corresponding to the <see cref="T:System.Uri" /> specified. Deleting a part that does not exist is not an error, so we do not throw an exception in that case.</summary>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackage.xml]:
```suggestion
        <summary>This method is for a custom implementation specific to the file format. Deletes the <see cref="T:System.IO.Packaging.PackagePart" /> corresponding to the <see cref="T:System.Uri" /> specified. Attempting to delete a part that does not exist is not an error and so does not throw an exception.</summary>
```

File: xml/System.IO.Packaging/ZipPackage.xml [hunk 14]
```diff
@@ -113,9 +116,11 @@
...
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">If the <paramref name="partUri" /> parameter is <see langword="null" />.</exception>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackage.xml]:
```suggestion
        <exception cref="T:System.ArgumentNullException"><paramref name="partUri" /> is <see langword="null" />.</exception>
```

File: xml/System.IO.Packaging/ZipPackage.xml [hunk 15]
```diff
@@ -113,9 +116,11 @@
...
+        <exception cref="T:System.ArgumentException">If the <paramref name="partUri" /> parameter does not have a valid <see cref="T:System.Uri" /> syntax.</exception>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackage.xml]:
```suggestion
        <exception cref="T:System.ArgumentException"><paramref name="partUri" /> does not have a valid <see cref="T:System.Uri" /> syntax.</exception>
```

File: xml/System.IO.Packaging/ZipPackage.xml [hunk 16]
```diff
@@ -206,8 +211,8 @@
         <Parameter Name="partUri" Type="System.Uri" />
       </Parameters>
       <Docs>
-        <param name="partUri">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="partUri">The <see cref="T:System.Uri" /> that represents the <see cref="T:System.IO.Packaging.PackagePart" />.</param>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackage.xml]:
```suggestion
        <param name="partUri">The Uri that represents the <see cref="T:System.IO.Packaging.PackagePart" />.</param>
```

File: xml/System.IO.Packaging/ZipPackage.xml [hunk 17]
```diff
@@ -237,8 +242,8 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>This method is for a custom implementation specific to the file format. This is the method that knows how to get the actual parts from the underlying <see cref="T:System.IO.Packaging.ZipPackage" />.</summary>
+        <returns>An array of <see cref="T:System.IO.Packaging.ZipPackagePart" />.</returns>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackage.xml]:
```suggestion
        <returns>An array of <see cref="T:System.IO.Packaging.ZipPackagePart" /> objects.</returns>
```

File: xml/System.IO.Packaging/ZipPackage.xml [hunk 18]
```diff
@@ -237,8 +242,8 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>This method is for a custom implementation specific to the file format. This is the method that knows how to get the actual parts from the underlying <see cref="T:System.IO.Packaging.ZipPackage" />.</summary>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackage.xml]:
```suggestion
        <summary>This method is for a custom implementation specific to the file format. This method retrieves the actual parts from the underlying <see cref="T:System.IO.Packaging.ZipPackage" />.</summary>
```

File: xml/System.IO.Packaging/ZipPackagePart.xml [hunk 19]
```diff
@@ -71,12 +71,11 @@
...
-        <param name="mode">To be added.</param>
-        <param name="streamFileAccess">To be added.</param>
-        <param name="access">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="mode">The <see cref="T:System.IO.FileMode" /> which the stream should be opened.</param>
+        <param name="access">The <see cref="T:System.IO.FileAccess" /> with which the stream should be opened.</param>
+
+        <summary>Custom implementation for the GetStream method.</summary>
+        <returns>Stream corresponding to this <see cref="T:System.IO.Packaging.ZipPackagePart" />.</returns>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackagePart.xml]:
```suggestion
        <returns>A stream corresponding to this <see cref="T:System.IO.Packaging.ZipPackagePart" />.</returns>
```

File: xml/System.IO.Packaging/ZipPackagePart.xml [hunk 20]
```diff
@@ -71,12 +71,11 @@
...
       </Parameters>
       <Docs>
-        <param name="streamFileMode">To be added.</param>
-        <param name="mode">To be added.</param>
-        <param name="streamFileAccess">To be added.</param>
-        <param name="access">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="mode">The <see cref="T:System.IO.FileMode" /> which the stream should be opened.</param>
+        <param name="access">The <see cref="T:System.IO.FileAccess" /> with which the stream should be opened.</param>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackagePart.xml]:
```suggestion
        <param name="access">The access permissions with which the stream should be opened.</param>
```

File: xml/System.IO.Packaging/ZipPackagePart.xml [hunk 21]
```diff
@@ -71,12 +71,11 @@
         <Parameter Name="access" Type="System.IO.FileAccess" Index="1" FrameworkAlternate="netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8" />
       </Parameters>
       <Docs>
-        <param name="streamFileMode">To be added.</param>
-        <param name="mode">To be added.</param>
-        <param name="streamFileAccess">To be added.</param>
-        <param name="access">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="mode">The <see cref="T:System.IO.FileMode" /> which the stream should be opened.</param>
```
Comment (rpetrusha) [inline xml/System.IO.Packaging/ZipPackagePart.xml]:
```suggestion
        <param name="mode">The I/O mode in which to open the stream.</param>
```

**Comment by rpetrusha** [comment]
This looks good, @carlossanlop. I'll merge your PR now.

**Pull request merged**